test/primitives.jl

```diff
@@ -22,6 +22,26 @@
     @test r(T(1.)) == P2(1,1)
     @test r(T(Inf)) == P2(Inf,Inf)
     @test_throws DomainError(T(-1), "r(t) is not defined for t < 0.") r(T(-1))
+    
+    p₁ = Point(3,3,3.)
+    p₂ = Point(-3,-3,-3.)
+    p₃ = Point(1,0,0)
+    r = Ray((0,0,0), (1,1,1.))
+    @test p₁ ∈ r
+    @test p₂ ∉ r
+    @test p₃ ∉ r
+
+
+    r1 = Ray((0,0,0), (1,0,0.))
+    r2 = Ray((1,1,1), (1,2,1.))
+    @test r1 != r2
+  
+    r1 = Ray((0,0,0), (1,0,0.))
+    r2 = Ray((1,0,0), (-1,0,0.))
+    @test r1 != r2
+  
+    r = Ray((0,0,0), (1,0,0.))
+    @test r == r
```

Comment:
Notice that all these tests need to be changed to use the T type as the other tests. @dorn-gerhard we are wasting a lot of time reviewing this PR because you are not following the style of the surrounding code. Can you please pay more attention in the next round of reviews?